packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx

```diff
@@ -153,7 +153,7 @@ export const Avatar: ComponentWithAs<'div', AvatarProps> & FluentComponentStatic
   setEnd();
 
   return result;
-};
+}) as unknown) as ForwardRefWithAs<'div', HTMLDivElement, AvatarProps> & FluentComponentStaticProps<AvatarProps>;
```

Comment:
this casting is only there to handle things like `create` and `handledProps` right ?

Reply:
In this case I think we should probably extend `React.FrowardExoticRefComponent` without own type to add the custom northstar properties

Reply:
mb something like:
```ts
type ForwardRefComponent<TElement, TProps> = React.ForwardRefExoticComponent<TProps & React.RefAttributes<TElement>> & {
  handledProps?: string[];
  create?: any;
};
```

we would also need to update `createShorthand` to support exotic forward ref components

Reply:
Issue to track it https://github.com/microsoft/fluentui/issues/20594